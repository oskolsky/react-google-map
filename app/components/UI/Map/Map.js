import React from 'react'
import PropTypes from 'prop-types'

import {
    GoogleMap,
    Marker,
    withGoogleMap,
    withScriptjs,
} from 'react-google-maps'

import styles from './Map.scss'

export default class Map extends React.Component {
    static propTypes = {
        center: PropTypes.shape({ latitude: PropTypes.number, longitude: PropTypes.number }),
        zoom: PropTypes.number,
    }

    static defaultProps = {
        zoom: 10,
    }

    render() {
        const center = {
            lat: this.props.center.latitude,
            lng: this.props.center.longitude,
        }

        const MapWithAMarker = withScriptjs(withGoogleMap(() =>
            <GoogleMap defaultCenter={center} defaultZoom={this.props.zoom}>
                <Marker position={center} label="I" />
            </GoogleMap>
        ))

        return (
            <div className={styles.base}>
                <MapWithAMarker
                    googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCX0yWmVXi9hNkpjAvw1Cfa8N0PV0zaeO8&v=3.exp&libraries=geometry,drawing,places"
                    loadingElement={<div style={{ height: '100%' }} />}
                    containerElement={<div style={{ height: '100%' }} />}
                    mapElement={<div style={{ height: '100%' }} />}
                />
            </div>
        )
    }
}
