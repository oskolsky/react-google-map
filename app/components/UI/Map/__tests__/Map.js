import React from 'react'
import { shallow } from 'enzyme'

import Map from '../Map'

function setup(options) {
    const props = {
        center: {
            latitude: 0,
            longitude: 0,
        },
        zoom: 10,
    }

    const enzymeWrapper = shallow(
        <Map {...props} {...options} />
    )

    return {
        props,
        enzymeWrapper,
    }
}

let enzymeWrapper = null

describe('Map specification', () => {
    afterEach(() => {
        enzymeWrapper.unmount()
    })

    it('renders correctly', () => {
        enzymeWrapper = setup().enzymeWrapper

        expect(enzymeWrapper).toMatchSnapshot()
    })
})
