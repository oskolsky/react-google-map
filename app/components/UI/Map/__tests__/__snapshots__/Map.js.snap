// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map specification renders correctly 1`] = `
<div
  className="base"
>
  <withScriptjs(withGoogleMap(Component))
    containerElement={
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      />
    }
    googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCX0yWmVXi9hNkpjAvw1Cfa8N0PV0zaeO8&v=3.exp&libraries=geometry,drawing,places"
    loadingElement={
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      />
    }
    mapElement={
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      />
    }
  />
</div>
`;
